### PR TITLE
Add Quickstart guide links to install instructions

### PR DESCRIPTION
**Pull request by mocofound**
Bringing the learn pages, quickstart and Nomad website in line with each other.

### CODE REVIEW

**Comment by mocofound** [comment]
I would still say these edits are still necessary or we may have some confusing logic here we need to iron out. In addition to the circular logic, I think we should make more of an effort to show how easy it is to install Nomad not only locally, but on your infrastructure or training platform of choice. @angrycub @yishan-lin 

The docs ["Overview"](https://nomadproject.io/docs/install) page looks like it has been linked to Learn for "Installing Nomad".  

**Should "Overview" actually look more like the commits in this PR (show an overview of possible docs/installation methods guides as opposed to install details)?**

The docs ["Quickstart"](https://nomadproject.io/docs/install/quickstart/) Page may now redundant as it is currently very similar to "Overview" 

**Should this actually be the page linked to Learn?**

The ["Main Docs Page"](https://nomadproject.io/docs/) doesn't make it very clear users should dive into the navigation on the left.  The link back to the "Intro" is featured prominently and takes the users out of "/docs". 

**Should we should include links to the "Overview and/or Quickstart" guide(s) on the "Main Docs Page"?**

Additionally, Intro ["Getting Started"](https://nomadproject.io/intro/getting-started/) now also links to the Learn page and seems redundant.

**Should we consolidate Intro "Getting Started" be consolidated with Docs "Overview/Quickstart"?**

TLDR, I request feedback on the following proposal:
* Modify this PR #6772 lines 21-23 to supplement and/or replace HashiCorp Learn links with Nomad Docs links (that link to Learn behind the scenes)
* Update ["Main Docs Page"](https://nomadproject.io/docs/) to include direct links to the sidebar ("QuickStart", "Overview", etc.) for folks looking for documentation in addition to directing new folks to "Intro".
* Merge content from this PR #6772 into Docs "Overview" or "Main Docs Page"
* Consolidate docs "Quickstart" with intro "Getting Started" to form a single docs "Quickstart/Getting Started" page 
* Remove any redundant pages docs ("Quickstart")

**Comment by notnoop** [comment]
Very sorry for taking so long to get back.  Closing this as it bitrotted with the ui refactor, and so much has changed since this PR is made. The AWS guides linked from this PR are super old at this point.  @angrycub will re-approach this issue with fresh take.  Thanks for your understanding.
